packages/connect-form/src/components/connect-form.tsx

```diff
@@ -187,8 +193,18 @@ function ConnectForm({
           </div>
           <div className={formFooterStyles}>
             <ConnectFormActions
+              initialConnectionInfo={initialConnectionInfo}
               errors={connectionStringInvalidError ? [] : errors}
               warnings={connectionStringInvalidError ? [] : warnings}
+              saveDisabled={!isDirty}
```

Comment:
just a thought: is a bit strange to derive whether to show the save button inside the  `ConnectFormActions` component but pass if it should be disabled directly. Not sure if this would improve it, but maybe something like `saveButton: 'enabled' | 'disabled' | 'hidden'`?

Reply:
change

Reply:
Looks good to me, if you feel this is weird we could also just pass everything to the component as is (`isDirty` and `favorite`) and do the "math" inside where you need it.